### DISCUSSION TITLE
feat: ellipsis on table cell

### DISCUSSION THREAD

**Discussion by mizek1** [iurynogueira/ion#759] [Ideas]
Sometimes we have a large information to show on a column, that makes a cell too large on both aspects, width and height:

![image](https://github.com/Brisanet/ion/assets/57760325/97f4957a-ee14-4907-85c3-047f643bb25f)

We need to be able to, when we set a width on a column, we can also set if that column could be partially hidden using ellipsis properties. We also need that, when we decide to use ellipsis on a column, the table automatically shows a tooltip with the complete information. Something like this:

![image](https://github.com/Brisanet/ion/assets/57760325/35a4296b-a8d0-41a0-aae4-c9ee2d3e127e)

**Comment by luccskywalker** [comment]
Qual o sentido de ter uma tabela que esconde o que eu estou mostrando nela? 
Esse comportamento me parece mais um acordion do que uma tabela.

**Comment by mizek1** [comment]
> Qual o sentido de ter uma tabela que esconde o que eu estou mostrando nela? 
> Esse comportamento me parece mais um acordion do que uma tabela. 

The default behavior is to show all information, this feat will only allow the possibility to hide part of the information that can be hidden. Every project has it's own specifications.

**Comment by luccskywalker** [comment]
I've forgotten only English in here. 
So, you will define the criteria to hide the content, how do you will control it. 
If these steps are not well specified it only will bring a behavior unnecessary to the component. 
The master proposition of a table is to show information, not to be pretty. Can be both obviously but if these two principles collides, i think that display the info should have more  weight on the criteria selection.

**Comment by mizek1** [comment]
"The master proposition of a table is to show information". Yes, it is, that is why we need the ellipsis on huge information, in the first place. On a page, we need that our user see more data on the available space and find the item they want. How do you feel, as a user, when you enter on a page and see this?
![image](https://github.com/Brisanet/ion/assets/57760325/ee74f7f8-75c4-42be-b58c-49a5309881dd)
I understand that a table doesn't need to be pretty, but we also need to garantee that a user **find the information in the fast way possible**, because that's why we use tables.

**Comment by mizek1** [comment]
What do you guys think of these solution? Any other idea to solve this problem?
@iurynogueira @andersonnagelo @Jeter-Megaron @qrayannes (I can't find Edney's user)

**Comment by luccskywalker** [comment]
If the user is not finding the information that him needs, the column is not representing their necessity.

**Comment by edneyql** [comment]
@mizek1 @luccskywalker  Sometimes the solution may not be a table, the solution could be related to another component... As I don't understand the context of the demand itself, I can't say so much if in this case it would be a table or not , but regarding the density of information, it also interferes with the user's cognitive process and loses the meaning of the component that serves to systematize the information, which must be presented in a clear and precise way, to facilitate the reading and understanding of the data. If I have a table that brings me such a high density of data and I can't deal logically with it, I have to rethink whether I'm using the right component for the solution.

**Comment by AlyssonMascarenhas** [comment]
@edneyql what if the table is just a way to extract through actions?

**Comment by AlyssonMascarenhas** [comment]
I believe we can create a discussion at github.com/Brisanet/ion/discussions instead of discussing it here in the issue.

**Comment by Jeter-Megaron** [comment]
The solution has already been aligned with the SIGA team, we are going to suppress the column information in the table, the user will be able to see all the data related to the selected profile with 1 additional click on the detailing column of the desired line.

**Comment by iurynogueira** [comment]
Guys, can follow with this implementation because can have other cases to use this abbreviation too. We need give other options to the dev and this improve will be more one.
